sklearn/tests/test_kernel_approximation.py

```diff
@@ -20,6 +21,54 @@
 Y /= Y.sum(axis=1)[:, np.newaxis]
 
 
+def test_polynomial_sampler():
+    # test that PolynomialSampler approximates polynomial
+    # kernel on random data
+
+    # compute exact kernel for degree=2
+    kernel = np.dot(X, Y.T)**2
+
+    # approximate kernel mapping
+    ps_transform = PolynomialSampler(n_components=1000, degree=2,
+                                     random_state=42)
+    X_trans = ps_transform.fit_transform(X)
+    Y_trans = ps_transform.transform(Y)
+    kernel_approx = np.dot(X_trans, Y_trans.T)
+
+    error = kernel - kernel_approx
+    assert np.abs(np.mean(error)) <= 0.01  # close to unbiased
```

Comment:
We usually use `assert_array_almost_equal(kernel, kernel_approx)`, would it be sufficient ? or is the approximation too coarse ?

Reply:
I am afraid so. I borrowed this from `test_rbf_sampler( )`.